Change event type and log type to structured enum

ruwi-next opened the pull request:
Hi

I think it would make a better interface if we used enums to represent the event type and log type. It also should be more efficient since we are using a pair of enum instead of a string.

We can also more accurately represent the possible states by nesting the log type in the enum variant where it is applicable.

This does make some assumptions about the combination of event type and log type but I think it is correct.

Any feedback is appreciated

puffyCid commented:
hey @ruwi-next thanks for the PR.
This seems reasonable. I merged some larger PRs before this one, could rebase and update the PR?
Will gladly merge after the conflicts are resolved

puffyCid commented:
sorry, for delay.
This PR removes the `log_type` field from `LogData`? It seems to merge it with the `EventType` field.
These should be separate fields similar to the `log` command

The log command labels entries similar to below
```
                log default, log error,
                activity create
                process signpost, thread signpost, system signpost
                trace default
```

ruwi-next commented:
No worries :)

> This PR removes the log_type field from LogData? It seems to merge it with the EventType field.

Yeah I was in two minds about combining or keeping those split. It felt like there were log types that were only used for certain event types but I did run into some issues when merging them. I've updated my branch to split them.

I've also updated my branch to rebase instead of use a merge commit as it looks like that is a requirement.

puffyCid reviewed (commented):
@ruwi-next looks good thanks!
Tests pass and clippy is happy

I left one comment if you could review
Happy to merge after that

puffyCid commented:
also it looks like the example file now fails to compile
```
error[E0308]: mismatched types
   --> unifiedlog_iterator/src/main.rs:376:21
    |
376 |                     record.event_type.to_owned(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `String`, found `EventType`

```

converting to strings should resolve
something like
```
                  format!("{:?}", record.event_type),
                    format!("{:?}", record.log_type),
```
If u can resolve that issue too, that would be great

puffyCid commented:
merged thanks!
